test_runner/src/main/kotlin/ftl/util/Utils.kt

```diff
@@ -128,3 +128,14 @@ fun <T> KMutableProperty<T?>.require() =
 fun getGACPathOrEmpty(): String = System.getenv("GOOGLE_APPLICATION_CREDENTIALS").orEmpty()
 
 fun saveToFlankLinks(vararg links: String) = File("flank-links.log").writeText(links.joinToString(System.lineSeparator()))
+
+fun getEnv(name: String): String? = System.getenv(name)
+
+fun List<String>.parseEnvsIfNeeded() = this
+    .partition { it.startsWith("$") }
+    .let { (envs, commonTestTargets) ->
+        commonTestTargets + envs
+            .mapNotNull { getEnv(it.drop(1)) }
+            .flatMap { it.split(",") }
+            .map { it.trim() }
+    }
```

Comment:
I guess it could be
```kotlin
fun List<String>.parseEnvsIfNeeded() = partition { it.startsWith("$") }
    .let { (envs, commonTestTargets) -> commonTestTargets + envs }
    .mapNotNull { getEnv(it.drop(1)) }
    .flatMap { it.split(",") }
    .map { it.trim() }
```

and then we avoid nested streams

Reply:
this function is used only in `test_runner.src.main.kotlin.ftl.args.CreateAndroidArgs|CreateIosArgs` so it would be better to put it somewhere there instead of publishing project-wide util

Reply:
Totally agree, thanks!

Reply:
> I guess it could be
> 
> ```kotlin
> fun List<String>.parseEnvsIfNeeded() = partition { it.startsWith("$") }
>     .let { (envs, commonTestTargets) -> commonTestTargets + envs }
>     .mapNotNull { getEnv(it.drop(1)) }
>     .flatMap { it.split(",") }
>     .map { it.trim() }
> ```
> 
> and then we avoid nested streams

I think that won't work, I need to parse strings with `$` differently. The proposed solution will spread the list into two and then combine them into one and perform mapping on the combined one.

But, as I wrote, I agree with your suggestion regarding package